src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,198 @@
+package datawave.query.util;
+
+import org.apache.accumulo.core.data.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This class handles the serialization and deserialization of the Accumulo value in a record of the Datawave Metadata table that has a column family of "f" and
+ * a column qualifier that is prefixed with the string "compressed-" like "compressed-csv" for example. This is a class used to help compress the date and
+ * frequency values that are aggregated together to by the FrequencyTransformIterator and manipulated in the FrequencyFamilyCounter. The byte array is in
+ * regular expression format ((YEAR)(4BYTE-FREQUENCY){366}))* . Explained verbally a four byte representation of Year followed by by up to 366 (Leap year) 4
+ * byte holders for frequency values. The month and day of the frequency value is coded by the position in the array. There aren't any delimiters between years
+ * and frequencies which adds to the compression. Each Accumulo row for this "aggregated" frequency "map" would be 10 x ( 4 + 4 + (366 * 4) ) bytes long for a
+ * maximum length for a 10 year capture of 14720 bytes.
+ */
+
+public class DateFrequencyValue {
+    
+    private static final Logger log = LoggerFactory.getLogger(DateFrequencyValue.class);
+    
+    public static final int DAYS_IN_LEAP_YEAR = 366;
+    private static int NUM_YEAR_BYTES = 4;
+    private static int NUM_FREQUENCY_BYTES = DAYS_IN_LEAP_YEAR * 4;
+    private static int NUM_BYTES_PER_FREQ_VALUE = 4;
+    
+    public DateFrequencyValue() {}
+    
+    /**
+     * @param dateToFrequencyValueMap
+     *            the keys should be dates in yyyyMMdd format
+     * @return size the size of the ByteArrayOutputStream
+     */
+    private int calculateOutputArraySize(TreeMap<YearMonthDay,Frequency> dateToFrequencyValueMap) {
+        int year, presentYear = 0, size = 0;
```

Comment:
How about returning an estimate which avoids iterating over the entire list:
```
int firstYear = dateToFrequencyValueMap.firstKey().year;
int lastYear = dateToFrequencyValueMap.lastKey().year;
return (lastYear - firstYear + 1) * (NUM_YEAR_BYTES + NUM_FREQUENCY_BYTES);
```
Or if you are so concerned about an exact length, then you can use the ceilingKey method to skip over all of the entries for a year.  I think an upper-bound calculation is fine for the purposes you are using this.